docs/manual.md

```diff
@@ -807,91 +805,294 @@ user:
+
+For example, kernel-param attribute returns a string, however, it can be tested using numeric comparisons:
+
+Example kernel-param test:
+```yaml
+kernel-param:
+  net.core.somaxconn:
+      value: "128"
+```
+
+Kernel-param (failing) test with transform:
+```yaml
+kernel-param:
+  net.core.somaxconn:
+      value: {gt: 200}
+```
+
+When a transformed test fails, it will detail the transformers used, the `-o include_raw` option can be used to include the raw, untransformed attribute value:
+```
+$ goss v
+F
+
+Failures/Skipped:
+
+KernelParam: net.core.somaxconn: value:
+Expected
+    128
+to be >
+    200
+the transform chain was
+    [{"to-numeric":{}}]
+
+Total Duration: 0.001s
+Count: 1, Failed: 1, Skipped: 0
+
+
+$ goss v -o include_raw
+F
 
-$ goss validate
-..
+Failures/Skipped:
+
+KernelParam: net.core.somaxconn: value:
+Expected
+    128
+to be >
+    200
+the transform chain was
+    [{"to-numeric":{}}]
+the raw value was
+    "128"
 
 Total Duration: 0.001s
-Count: 2, Failed: 0
+Count: 1, Failed: 1, Skipped: 0
+
 ```
 
-## Advanced Matchers
-Goss supports advanced matchers by converting json input to [gomega](https://onsi.github.io/gomega/) matchers.
+### Advanced Matchers
 
-### Examples
+Goss supports advanced matchers by converting YAML input to [gomega](https://onsi.github.io/gomega/) matchers.
+
+#### String Matchers
 
-Validate that user `nobody` has a `uid` that is less than `500` and that they are **only** a member of the `nobody` group.
+These will convert the system attribute to a string prior to matching.
 
+* `'55'` - Checks that the numeric is "55" when converted to string
+* `have-prefix: pre` - Checks if string starts with "pre"
+* `have-suffix: suf` - Checks if string ends with "suf"
+* `match-regexp: '.*'` - Checks if string matches regexp
+* `contain-substring: '2'` - Checks if string contains "2"
+
+Example:
 ```yaml
-user:
-  nobody:
-    exists: true
-    uid:
-      lt: 500
-    groups:
-      consist-of: [nobody]
+matching:
+  example:
+    content: 42
+    matches:
+      and:
+        - '42'
+        - have-prefix: '4'
+        - have-suffix: '2'
+        - match-regexp: '\d{2}'
+        - contain-substring: '2'
 ```
 
-Matchers can be nested for more complex logic, for example you can ensure that you have 3 kernel versions installed and none of them are `4.1.0`:
+#### Numeric matchers
 
+These will convert the system attribute to a numeric prior to matching.
+
+* `42` - If the expected type is a number
+* `gt, ge, lt, le` - Greater than, greater than or equal, less than, etc..
+
+Example:
 ```yaml
-package:
-  kernel:
-    installed: true
-    versions:
+matching:
+  example:
+    content: "42"
+    matches:
+      and:
+        - 42
+        - 42.0
+        - gt: 40
+        - lt: 45
+```
+
+#### Array matchers
+
+These will convert the system attribute to an array prior to matching. Strings are split on "\n"
+
```

Comment:
```suggestion
```

Reply:
again, this looks like `-` being suggested as an addition.. is that correct?